kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/commands/optaplanner/UpdateSolverStateCommand.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.api.commands.optaplanner;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import org.kie.server.api.model.KieServerCommand;
+import org.kie.server.api.model.instance.SolverInstance;
+
+import javax.xml.bind.annotation.*;
+
+@XmlRootElement(name = "update-solver-state")
+@XStreamAlias( "update-solver-state" )
+@XmlAccessorType(XmlAccessType.NONE)
+public class UpdateSolverStateCommand
+        implements KieServerCommand {
+
+    private static final long serialVersionUID = -1803374525440238478L;
+
+    @XmlAttribute(name = "container-id")
+    @XStreamAlias("container-id")
+    private String containerId;
+
+    @XmlAttribute(name = "solver-id")
+    @XStreamAlias("solver-id")
+    private String solverId;
+
+    @XmlElement
+    @XStreamAlias("solver-instance")
+    private String instance;
```

Comment:
Because the instance may contain user-provided classes (for instance, the problem to solve). This requires that attribute to be unmarshalled with the kie-container classloader, not the kie-server classloader. Unfortunately, this is the only way we found to do it. It is the same artifice drools/jbpm use.